### ISSUE TITLE
Impossible to add SelenideElement to ArrayList on Android

### ISSUE THREAD

**Issue by dkorobtsov** [codeborne/selenide#495]
Suppose just the same root cause as in: https://github.com/codeborne/selenide/issues/494
Another case where it causes a headache.

Scenario is following:

Page Object fields initialized:
```
    private SelenideElement addNumberTitle;
    private SelenideElement countryFlag;

    addNumberTitle = $(By.id(APP_ID + "buy_number_tv_title"));
    countryFlag = $(By.id(APP_ID + "buy_number_iv_country_flag"));

```

and afterwards used in ArrayList along with conditions:

```
    @Override
    public final ArrayList<UiRequirement> getUiRequirements() {

        log.debug("Providing " + getName() + " UI elements list for validation.");
        ArrayList<UiRequirement> requirements = new ArrayList<>();
        requirements.add(new UiRequirement(addNumberTitle, visible));
        requirements.add(new UiRequirement(countryFlag, visible));
```

UiRequirement is pretty simple:

```
public class UiRequirement {

    private SelenideElement mElementId;
    private Condition mCondition;

    public UiRequirement(SelenideElement elementId, Condition condition){

        mElementId = elementId;
        mCondition = condition;

    }
    public SelenideElement getElementId(){return mElementId; }

    public Condition getCondition(){return mCondition; }
}
```

Tried lots of ways to add SelenideObject to Array, all are causing exceptions. Debug output is below:

![image](https://cloud.githubusercontent.com/assets/13999072/23974521/5094b35e-09e4-11e7-97c0-0aa34e47e489.png)

At the same time, changing SelenideElement to By solves problem completely....

**Comment by asolntsev** [comment]
@dkorobtsov Actually it only throws `UnsupportedOperationException` when you debug it. In real life, it will not fail. It's a Heisenberg `toString()` :)

Fixed in Selenide 4.4

**Comment by dkorobtsov** [comment]
@asolntsev hmm... I've spent all evening yesterday trying to make it work, it was crashing on that line all the time.... :-( 

I'll test changes on branch where I changed all PO fields to SE and post an update.
